src/lib/network_pool/indexed_pool.ml

```diff
@@ -360,8 +360,13 @@ let handle_committed_txn :
       in
       if Account_nonce.(nonce_to_remove <> first_nonce) then
         failwith
-          "Tried to handle a committed transaction in the pool but its nonce \
-           doesn't match the head of the queue for that sender."
+          (sprintf
+             !"Tried to handle a committed transaction in the pool but its \
+               nonce doesn't match the head of the queue for that sender. \
+               Committed transaction: %{sexp: \
+               User_command.With_valid_signature.t} Head of the queue: \
+               %{sexp: User_command.With_valid_signature.t}"
+             committed first_cmd)
```

Comment:
Would it be better here to log an error through the Logger here with the extra debug information in metadata instead of sexping it into the user displayed message? If you do that, you could also add the entire queue's contents to the metadata potentially as well, helping us better debug if this triggers in the future.